src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1753,11 +1753,17 @@ internal static Win32ResourceForm DetectWin32ResourceForm(Stream win32Resources)
             return resourceList;
         }
 
-        internal void SetupWin32Resources(CommonPEModuleBuilder moduleBeingBuilt, Stream? win32Resources, DiagnosticBag diagnostics)
+        internal void SetupWin32Resources(CommonPEModuleBuilder moduleBeingBuilt, Stream? win32Resources, bool useRawWin32Resources, DiagnosticBag diagnostics)
```

Comment:
Rather than passing the `Stream` and `bool` separately here did you consider passing them around as a tuple? Essentially linking them together a bit harder. Possibly even making it a type and moving the logic for `DetectWin32ResourceForm`, etc ... up into that type? Essentially making it an abstraction to itself.

Reply:
Tomas and I discussed a bit how resources could be abstracted better, but I am reluctant to spend much time on it, because I feel like a *great* design here for an internal API isn't needed to reach our goals as long as it is decently sustainable (not prototype/hack mode), and I want to keep focused on getting the tool fully functioning.

We don't normally say stuff like that on the compiler team, but that's how it looks from where I'm standing 😅. Happy to revisit this if you feel differently.